tests/test_code_hasher.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+import threading
+
+from bionic.code_hasher import CodeHasher, TypePrefix
+
+
+def test_code_hasher():
+    def barray(value):
+        return bytearray(value, "utf8")
+
+    circular_dict_1_a = {"k11": "v1"}
+    circular_dict_2_a = {"k21": "v2"}
+    circular_dict_3_a = {"k31": "v3"}
+    circular_dict_1_a["k12"] = circular_dict_2_a
+    circular_dict_2_a["k22"] = circular_dict_3_a
+    circular_dict_3_a["k32"] = circular_dict_1_a
+
+    circular_dict_1_b = {"k11": "v1"}
+    circular_dict_2_b = {"k21": "v2"}
+    circular_dict_3_b = {"k31": "v3"}
+    circular_dict_1_b["k12"] = circular_dict_2_b
+    circular_dict_2_b["k22"] = circular_dict_3_b
+    circular_dict_3_b["k32"] = circular_dict_2_b
+
+    def f1():
+        v = 10
+        return v
+
+    def f2():
+        v = 20
+        return v
+
+    def f3():
+        v = "10"
+        return v
+
+    def f4():
+        return "10"
+
+    def inc(x):
```

Comment:
Should we test functions that have a wider range of constant values?

Reply:
Also, should we be testing functions with default values? What about functions containing more complex constructions?